web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1288,75 +1329,73 @@ namespace com.keyman.keyboards {
       const URL='https://api.keyman.com/cloud/4.0/'
                 + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards');
 
-      let promise = new Promise(function(resolve: (result: KeyboardStub[]) => void, reject: (error: Error) => void) {
-        const Lscript: HTMLScriptElement = keymanweb.util._CreateElement('script');
-
-        const queryConfig = '?jsonp=keyman.register&languageidtype=bcp47&version='+keymanweb['version'];
-
-        // Set callback timer
-        const timeoutID = window.setTimeout(function() {
-          delete kbdManager.registrationResolvers[timeoutID];
-          reject(new Error(CLOUD_TIMEOUT_ERR));
-        } ,10000);
-
-        // Save the resolve / reject functions.
-        kbdManager.registrationResolvers[timeoutID] = {
-          resolve: resolve,
-          reject: reject
-        };
-
-        const tFlag='&timerid='+ timeoutID;
+      try {
```

Comment:
Why is this new outer `try`-`catch` necessary?  I don't see any code necessitating it, though it's possible I've missed something.

Reply:
I was getting uncaught errors, but turns out this try catch didn't help. Reverted.